pkg/network/ebpf/c/prebuilt/tracer.c

```diff
@@ -423,6 +423,46 @@ int kprobe__tcp_set_state(struct pt_regs* ctx) {
     return 0;
 }
 
+SEC("kprobe/tcp_connect")
+int kprobe__tcp_connect(struct pt_regs* ctx) {
+    struct sock* sk = (struct sock*)PT_REGS_PARM1(ctx);
+    u64 pid_tgid = bpf_get_current_pid_tgid();
+
+    bpf_map_update_elem(&tcp_connect_sock, &pid_tgid, &sk, BPF_ANY);
+
+    return 0;
+}
+
+SEC("kretprobe/tcp_connect")
+int kretprobe__tcp_connect(struct pt_regs* ctx) {
+    u64 pid_tgid = bpf_get_current_pid_tgid();
+    log_debug("kretprobe/tcp_connect: tgid: %u, pid: %u\n", pid_tgid >> 32, pid_tgid & 0xFFFFFFFF);
+
+    // We only trace if the connection worked and is established
+    int rc = PT_REGS_RC(ctx);
+    if (rc != 0) {
+        return 0;
```

Comment:
You should clear the `tcp_connect_sock` map on the failure path too.

Reply:
Here is a good pattern of what to do: https://github.com/DataDog/datadog-agent/blob/e0468e06d9f58ebd04c38da75a7129dff0f7002b/pkg/network/ebpf/c/tcp-recv.h#L77-L91

Reply:
@brycekahle should we be using a map value after deleting it?

Reply:
We dereference it, so it will be safe.

Reply:
Should be better now.